ReactiveCocoa/NSObject+Lifetime.swift

```diff
@@ -15,10 +15,53 @@ extension Reactive where Base: NSObject {
 			let token = Lifetime.Token()
 			let lifetime = Lifetime(token)
 
+			let objcClass: AnyClass = (base as AnyObject).objcClass
+			let deallocSelector = sel_registerName("dealloc")!
+
+			// Swizzle `-dealloc` so that the lifetime token is released at the
+			// beginning of the deallocation chain, and only after the KVO `-dealloc`.
+			objc_sync_enter(objcClass)
+			if objc_getAssociatedObject(objcClass, &lifetimeKey) == nil {
+				objc_setAssociatedObject(objcClass, &lifetimeKey, true, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+
+				var existingImpl: IMP? = nil
+
+				let newImplBlock: @convention(block) (UnsafeRawPointer) -> Void = { objectRef in
+					_rac_objc_setAssociatedObject(objectRef, &lifetimeTokenKey, nil, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
```

Comment:
Would it make more sense to close over a local `Lifetime.Token?` variable instead of using an associated object here? If we're explicitly deallocating, then we don't seem to get any actual benefit from storing it as an associated object.